src/Utils/Jalasoft.TeamUp.Projects.ProjectsException/ProjectsException.cs

```diff
@@ -1,32 +1,55 @@
-﻿using System;
-
-namespace Jalasoft.TeamUp.Projects.ProjectsException
+﻿namespace Jalasoft.TeamUp.Projects.ProjectsException
 {
+    using System;
+    using FluentValidation;
+    using Microsoft.AspNetCore.Mvc;
+
     public class ProjectsException : Exception
     {
-        public enum ProjectsErros
+        public ProjectsException(ProjectsErrors code)
         {
-            InternalServerError = 500,
-            NotFound = 404,
+            this.ErrorMessage = new ErrorMessage();
+            this.ErrorMessage.Message = "The resource couldn't be found";
+            this.StatusCode = (int)ProjectsErrors.NotFound;
+            this.Error = new ObjectResult(this.ErrorMessage);
+            this.Error.StatusCode = this.StatusCode;
         }
-        public string ProjectsErrorMessage { get; set; }
-        public int StatusCode { get; set; }
 
-        public ProjectsException(ProjectsErros code)
+        public ProjectsException(ProjectsErrors code, Exception exception)
         {
+            this.ErrorMessage = new ErrorMessage();
+
             switch (code)
             {
-                case ProjectsErros.InternalServerError:
-                    this.StatusCode = (int) ProjectsErros.InternalServerError;
-                    this.ProjectsErrorMessage = "{\"message\": \"Something went wrong, please contact the TeamUp administrator.\"}";
+                case ProjectsErrors.BadRequest:
+                    var validationException = (ValidationException)exception;
+                    this.ErrorMessage.Message = "Please review the errors, inconsistent data.";
+                    this.ErrorMessage.Errors = validationException.Errors;
+                    this.StatusCode = (int)ProjectsErrors.BadRequest;
+                    this.Error = new ObjectResult(this.ErrorMessage);
+                    this.Error.StatusCode = this.StatusCode;
+                    break;
+                case ProjectsErrors.InternalServerError:
+                    this.ErrorMessage.Message = "Something went wrong, please contact the TeamUp administrator.";
+                    this.StatusCode = (int)ProjectsErrors.InternalServerError;
+                    this.Error = new ObjectResult(this.ErrorMessage);
+                    this.Error.StatusCode = this.StatusCode;
                     break;
-                case ProjectsErros.NotFound:
-                    this.StatusCode = (int) ProjectsErros.NotFound;
-                    this.ProjectsErrorMessage = "{\"message\": \"The resource couldn't be found.\"}";
+                case ProjectsErrors.NotFound:
```

Comment:
having not found in the other constructor do we still need it here? is there a use case for this error to be here?

Reply:
Sorry, forgot to remove